icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -211,18 +284,40 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         int correctedNumBreaks = 0;
         for (int i = numBreaks - 1; i >= 0; i--) {
             int pos = charPositions[t_boundary[i]] + startPos;
-            if (!(foundBreaks.contains(pos) || pos == startPos)) {
+            if (!foundBreaks.contains(pos)
+                    && (pos != startPos
+                    || (isPhraseBreaking && pos == startPos && pos > 0 && isClosePunct(
+                    inText.setIndex(pos - 1))))) {
                 foundBreaks.push(charPositions[t_boundary[i]] + startPos);
                 correctedNumBreaks++;
             }
         }
 
         if (!foundBreaks.isEmpty() && foundBreaks.peek() == endPos) {
-            foundBreaks.pop();
-            correctedNumBreaks--;
+            // In phrase breaking, there has to be a breakpoint between Cj character and
+            // number/open punctuation e.g. １枚▁４０００円, 文字▁「そうだ...」
+            if (isPhraseBreaking) {
+                if (!isOpenPunct(inText.setIndex(endPos)) && !fNumberWordSet.contains(
```

Comment:
wrap the line before && to make it easier to read

Reply:
Resolved